crates/yulgen/src/mappers/expressions.rs

```diff
@@ -272,13 +272,26 @@ pub fn expr_unary_operation(context: &mut FnContext, exp: &Node<fe::Expr>) -> yu
     if let fe::Expr::UnaryOperation { op, operand } = &exp.kind {
         let yul_operand = expr(context, operand);
 
+        let typ = &context
+            .expression_attributes(operand)
+            .expect("Missing `operand` expression in context")
+            .typ;
+
         return match &op.kind {
             fe::UnaryOperator::USub => {
                 let zero = literal_expression! {0};
                 expression! { sub([zero], [yul_operand]) }
             }
             fe::UnaryOperator::Not => expression! { iszero([yul_operand]) },
-            _ => todo!(),
+            fe::UnaryOperator::Invert => match typ {
+                Type::Base(Base::Numeric(integer)) if integer.size() < 32 => {
+                    expression! { [names::adjust_numeric_size(integer)]((not([yul_operand]))) }
```

Comment:
Oh, we should also have operations for these adjustment functions. This way we are constructing calls to these functions in a single place.

e.g. 

replace

```rust
{
  expression! { [names::adjust_numeric_size(integer)]((not([yul_operand]))) }
}
```

with

```
numeric_operations::adjust_numeric_size(integer, yul_operand)
```

Reply:
:+1: done